### ISSUE TITLE
compound/flat inconsistency

### ISSUE THREAD

**Issue by tzshi** [UniversalDependencies/UD_English-EWT#59]
When I looked into cases where a word has both `compound` and `flat` children, I see some seemingly-inconsistent annotations. 

Examples from Grew: http://match.grew.fr/?custom=5ace8a92b2c1f&corpus=UD_English

We have for example:
... Prime <-[compound]- Minister -[flat]-> Iyad Allawi ...
... U.S. <-[compound]- president -[flat]-> George Bush ...

and we also have:
... Leader <-[compound]- Mahmoud -[flat]-> Abbas ...
... adviser  <-[compound]- Karen -[flat]-> Hughes ...

**Comment by sebschu** [comment]
From looking at a few of these examples, it seems like all of them should be annotated like the first two examples that you provide (like, e.g., _Prime Minister Iyad Allawi_). 

Pull requests for fixing inconsistencies are always welcome :)

**Comment by sylvainkahane** [comment]
I never understood why _President Obama_ should be `flat`. The main and simple test to decide whether a construction AB is headed is to look whether AB can be replaced only by A or only by B.

Clearly _President Obama_ can be replaced by _Obama_ and not by _President_:

> President obama cam to Paris
> Obama came to Paris
> *President came to paris

So this seems to be the common English noun-noun construction, which is analyzed as `compound` in other cases. Same thing for _Prime Minister Iyad Allawi, French actor Gaspard Ulliel_, _Leader Mahmoud Abbas_, or _Mister Miller._ 

 `flat` is only for headless constructions.

**Comment by dseddah** [comment]
in that case, Prime Minister Iyad Allawi, flat is used  because both are possible as head [Prime Minister] or [Iyad Allawy], no ?

**Comment by sebschu** [comment]
@dseddah Well, but you still cannot say _"Prime Minister came to the meeting"_ without adding a determiner.

@sylvainkahane I personally totally agree with your analysis but others don't. See, e.g., https://github.com/UniversalDependencies/docs/issues/324.

**Comment by sylvainkahane** [comment]
@sebschu Thanks for the answer. 

[UniversalDependencies/docs#324](https://github.com/UniversalDependencies/docs/issues/324) was about `name`, which no longer exists. Now UD has moved to more syntactic criteria and `flat` is clearly defined as a relation for headless constructions. Today, nothing justifies to use `flat` for _President Obama_, except for the history of UD.

**Comment by dan-zeman** [comment]
I agree with @sylvainkahane that in _President Obama_, _Obama_ is the head **in English** (because you cannot just remove _Obama_ without adding an article). However, it is not necessarily true about corresponding phrases in other languages. For example, Czech does not have articles, and you can replace the phrase _prezident Obama_ by either _prezident_ or _Obama_. Therefore, the phrase is headless in Czech, and suitable for the `flat` relation.

Unfortunately, people sometimes apply rules from one language to translated phrases in another language without noticing that the syntactic environment has changed. Names and titles are particularly prone to this sort of misguidance because they often look very similar in the two languages. We had the problem with `name` and we apparently still have it with `flat`, so we should make sure that the documentation makes the cross-language differences very clear, with examples from multiple languages.

**Comment by amir-zeldes** [comment]
Hi all, I just wanted to point out that if we take "President X" out of flat, then the same analysis should be applied to "Mr." and other titles in English. Is that something we want for languages with English-like determiners? I think it would affect Romance and Germanic as well.

Note also that there can be a sequence of titles, in which case I'm not sure about their internal headedness (e.g. German "Frau Professor Doktor X"). In languages with strong case marking I think there's usually agreement (e.g. Polish "pana direktora" - Mr./gen. director/gen.), so it's hard to motivate what's governing what.

**Comment by dan-zeman** [comment]
In the German example, you could probably do

    flat(Frau, Professor)
    flat(Frau, Doktor)
    nmod(X, Frau)

In Polish, I suppose it should be

    flat(pana, direktora)

**Comment by nschneid** [comment]
@amir-zeldes Could it be that German is reusing the adjectival modification construction for titles like *Herr*? Or are there tests for attributive adjectives that would fail?

It seems that different languages have recruited different core-grammar constructions for the formation of names, so maybe we can make the commonalities explicit with cross-cutting subtypes. E.g. English could have `compound:pre-name` whereas German could have `amod:pre-name` for titles like *Mr.*/*Herr*.

**Comment by dan-zeman** [comment]
_Herr_ is `NOUN`, definitely not `ADJ`. Therefore, if it is `*mod`, then ti must be `nmod`, nod `amod`.

**Comment by amir-zeldes** [comment]
Yes, German adjectives have distinctive morphology, this is definitely a noun

**Comment by nschneid** [comment]
We have settled on `nmod:desc` for this construction: https://universaldependencies.org/en/dep/nmod-desc.html

**Comment by nschneid** [comment]
- [ ] [EWT NOUN attached as compound, should be nmod:desc](https://universal.grew.fr/?custom=6769fe5d4b8b1) (only 1 token where it is flat)
- [ ] [GUM NOUN at beginning of flat expression, should be nmod:desc](https://universal.grew.fr/?custom=6769ff375aa6e)

**Comment by nschneid** [comment]
@amir-zeldes
